docs/README.md

```diff
@@ -0,0 +1,114 @@
+<!-- 
+Reference Links:
+ -->
+[core-contracts-repo]: https://github.com/Layr-Labs/eigenlayer-contracts
+[core-docs-m2]: https://github.com/Layr-Labs/eigenlayer-contracts/tree/m2-mainnet/docs
+
+[core-registerToAVS]: https://github.com/Layr-Labs/eigenlayer-contracts/blob/m2-mainnet/docs/core/DelegationManager.md#registeroperatortoavs
+[core-deregisterFromAVS]: https://github.com/Layr-Labs/eigenlayer-contracts/blob/m2-mainnet/docs/core/DelegationManager.md#deregisteroperatorfromavs
+
+## EigenLayer Middleware Docs
+
+EigenLayer AVSs ("actively validated services") are protocols that make use of EigenLayer's restaking primitives. AVSs are validated by EigenLayer operators, who are backed by delegated restaked assets via the [EigenLayer core contracts][core-contracts-repo]. Each AVS will deploy or modify instances of the contracts in this repo to hook into the EigenLayer core contracts and ensure their service has an up-to-date view of its currently-registered operators.
+
+**Currently, each AVS needs to implement one thing on-chain:** registration/deregistration conditions that define how an operator registers for/deregisters from the AVS. This repo provides building blocks to support these functions.
+
+*Eventually,* the core contracts and this repo will be extended to cover other conditions, including:
+* payment conditions that define how an operator is paid for the services it provides
+* slashing conditions that define "malicious behavior" in the context of the AVS, and the punishments for this behavior
+
+*... however, the design for these conditions is still in progress.*
+
+**Additional Note**: Although the goal of this repo is to eventually provide general-purpose building blocks for all kinds of AVSs, many of the contracts and design considerations were made to support EigenDA. When these docs provide examples, they will often cite EigenDA.
```

Comment:
might be a bit of a rabbit hole for people to fall down, but you _could_ link to the EigenDA repo here?

Reply:
[ac38452](https://github.com/Layr-Labs/eigenlayer-middleware/pull/96/commits/ac3845212bca0232f3db26882c12568b4a6e3660)